step-01/js/main.js

```diff
@@ -1,26 +1,73 @@
 'use strict';
 
-navigator.getUserMedia = navigator.getUserMedia ||
-    navigator.webkitGetUserMedia || navigator.mozGetUserMedia;
+// Define navigator.mediaDevices.getUserMedia() for older devices.
 
-var constraints = {
+// Old browsers might not implement mediaDevices, so we set an empty object.
+if (navigator.mediaDevices === undefined) navigator.mediaDevices = {};
+
+// Add mediaDevices.getUserMedia method if it's missing.
+if (navigator.mediaDevices.getUserMedia === undefined) {
+
+  /**
+   * Gets a stream of media content containing tracks of video or audio.
+   * Produces a Promise which resolves into a MediaStream object.
+   * @param {!MediaStreamConstraints} constraints to apply to the media stream.
+   * @return {!Promise} Promise which resolves to a MediaStream.
+   */
+  navigator.mediaDevices.getUserMedia = (constraints) => {
+    // Leverage legacy getUserMedia, if it is defined.
+    const getUserMedia = navigator.getUserMedia ||
+        navigator.webkitGetUserMedia || navigator.mozGetUserMedia;
+
+    // If legacy method is also not defined, resolve Promise as an error.
+    if (!getUserMedia) {
+      return Promise.reject(
+        new Error('getUserMedia is not implemented in this browser'));
+    }
+
+    // Otherwise, wrap the call to the navigator.getUserMedia with a Promise.
+    return new Promise(function(resolve, reject) {
+      getUserMedia.call(navigator, constraints, resolve, reject);
+    });
+  }
+}
+
+// Codelab starts here.
+
+/**
+ * Constraints to apply to MediaStream.
+ * On this codelab, we will be streaming video, but not audio.
+ * @type {!MediaStreamConstraints}
+ */
+const mediaStreamConstraints = {
   audio: false,
-  video: true
+  video: true,
 };
 
-var video = document.querySelector('video');
+/** @type {!Element} Video element where stream will be placed. */
+const localVideo = document.querySelector('video');
 
-function successCallback(stream) {
-  window.stream = stream; // stream available to console
+/**
+ * Handles success by adding the MediaStream to the video element.
+ * @param {!MediaStream} mediaStream generated by getUserMedia to add to video.
+ */
+const localMediaStreamSuccessCallback = (mediaStream) => {
+  window.stream = stream;  // Makes stream available to the console.
   if (window.URL) {
```

Comment:
Can get rid of the extra stuff now that video.srcObject is supported by the target browsers (and I'd prefer a simpler function name): 

```
function handleSuccess(stream) {
  video.srcObject = stream;
}
```

Reply:
Done

Reply:
I simplified function names, but kept the "local" part of the name.

The idea is to build up on top of each step on the next one and keep the same function names and code so you can focus only on the increment.